lib/metasploit_payloads/mettle.rb

```diff
@@ -52,19 +56,19 @@ def generate_argv
     def short_opt(opt)
       case opt
       when :background
-        'b'
+        'b '
```

Comment:
It looks like `:background` gets a space where `:debug` does not, even though the code that parses the options is nearly identical. What did you see that lets this tolerate a space and not `:debug`?

Reply:
The getopt parser does not like spaces between the flag and int values.  Debug is the only value that takes an int.

Reply:
Unless I am reading something wrong, so can `-b`: https://github.com/rapid7/mettle/blob/master/mettle/src/main.c#L120

Reply:
Good point.  My best answer is that the debug option does not work if there's a space.  For background, the changes here are functionally equivalent to how it was before the PR; rather than add the space in the case statement, it is done so in the command line creation.  As the background options does not appear to be supported on the framework side yet, I can't tell you if it does or doesn't work definitively.

Reply:
Note, both options take an optional parameter, which is why they don't use the space. '-b' can take a '0' or a '1', and if you don't specify a value, it assumes '1'. Same thing for '-d' in fact, it's just that log level '1' isn't very chatty out of the box. I've obviously over-complicated this interface a little because I wanted the ability to override baked-in values inside of a binary from the command line.

Reply:
Not at all; in fact I strongly endorse it having learned this lesson! See below.